src/ln/channelmonitor.rs

```diff
@@ -1247,6 +1247,42 @@ impl ChannelMonitor {
 
 			log_trace!(self, "Got broadcast of non-revoked remote commitment transaction {}", commitment_txid);
 
+			// TODO: We really should only fail backwards after our revocation claims have been
+			// confirmed, but we also need to do more other tracking of in-flight pre-confirm
+			// on-chain claims, so we can do that at the same time.
+			if let Storage::Local { ref current_remote_commitment_txid, ref prev_remote_commitment_txid, .. } = self.key_storage {
+				if let &Some(ref txid) = current_remote_commitment_txid {
+					if *txid != commitment_txid {
```

Comment:
Not sure of *txid != commitment_txid" ? I mean I can get it out and tests still pass ?

Reply:
Grrr, good point. I'll work on adding a test that hits it tomorrow. I did refactor this code to catch some other issues as noted above, but you can still comment out the is_tx_were_processing check and the tests pass.

Reply:
Ah, ok, so technically this could be hit if the dust limit were allowed to change after a channel was created (which it is not), allowing an HTLC to exist in both commitment transactions but be dust in one and non-dust in another. I removed the checks and added a big comment.

Reply:
Okay, not sure that the second part of comment clarify things, I mean "we'd want to be check whether *any* broadcastable commitment transaction has the HTLC in it", dust limit being a moving target, which inclusion would be authoritative to qualify a HTLC as below-dust? (but okay that's a comment on something non-existent so..)

Reply:
If dust limit were dynamic we wouldn't have an "authoritative" commitment transaction - it being non-dust in any broadcastable transaction would suffice to prevent us from treating it as dust here.